samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+   * in which case you can leave this value `null`. If the number of visible items is always one
+   * or two, it might make sense to just set this to the larger of the two to reduce churn in the
+   * preloader.
+   * @param requestBuilderModifier See [ListPreloader.PreloadModelProvider.getPreloadRequestBuilder]
+   */
+  // TODO(judds): Consider wrapping a LazyRow / LazyColumn and providing state instead of a separate
+  // function. Wrapping might also make it easier to pass through the size and request builder
+  // modifications so that it's easier to make sure the preload size matches a size on the
+  // GlideImage
+  @Composable
+  fun <DataTypeT : Any> GlideLazyListPreloader(
+    state: LazyListState,
+    data: List<DataTypeT>,
+    size: Size,
+    numberOfItemsToPreload: Int,
+    fixedVisibleItemCount: Int? = null,
+    requestBuilderModifier:
+      (data: DataTypeT, requestBuilder: RequestBuilder<*>) -> RequestBuilder<*>,
+  ) {
+    Log.d("TEST", "GlidePreloader")
+    val preloader =
+      rememberGlidePreloader(
+        data = data,
+        size = size,
+        numberOfItemsToPreload = numberOfItemsToPreload,
+        requestBuilderModifier = requestBuilderModifier
+      )
+    doPreload(preloader = preloader, state = state, fixedVisibleItemCount = fixedVisibleItemCount)
+  }
+
+  @Composable
+  private fun <DataTypeT: Any> doPreload(
+    preloader: ListPreloader<DataTypeT>, state: LazyListState, fixedVisibleItemCount: Int?
+  ) {
+    Log.d("TEST", "doPreload")
+    val lazyListVisibleInfo = derivedStateOf { state.lazyListVisibleInfo(fixedVisibleItemCount) }
+    // TODO(judds): Without this remember call, we're never recomposed. It's unclear what the right
+    // way to do this is?
+    remember(lazyListVisibleInfo.value) {}
+    SideEffect {
+      preloader.onScroll(
+        null,
+        lazyListVisibleInfo.value.firstVisibleItemIndex,
+        lazyListVisibleInfo.value.visibleItemCount,
+        lazyListVisibleInfo.value.totalItemCount)
+    }
+  }
+
+  @Composable
+  private fun <DataTypeT: Any> rememberGlidePreloader(
+    data: List<DataTypeT>,
+    size: Size,
+    numberOfItemsToPreload: Int,
+    requestBuilderModifier:
+      (data: DataTypeT, requestBuilder: RequestBuilder<*>) -> RequestBuilder<*>,
+  ): ListPreloader<DataTypeT> {
+    Log.d("TEST", "rememberGlidePreloader")
+    val currentData = remember(data) { data }
+    val requestManager = Glide.with(LocalContext.current)
+    class PreloadModelProvider : ListPreloader.PreloadModelProvider<DataTypeT> {
+      var currentData: List<DataTypeT> = emptyList()
+      override fun getPreloadItems(position: Int): MutableList<DataTypeT> {
+        return mutableListOf(this.currentData[position])
+      }
+
+      override fun getPreloadRequestBuilder(item: DataTypeT): RequestBuilder<*> {
+        return requestBuilderModifier(item, requestManager.asDrawable())
+      }
+    }
+
+    val preloadModelProvider = remember { PreloadModelProvider() }
```

Comment:
Because this class captures parameters from the composable function, move this to an anonymous class (`object :`) inside the `remember` where you instantiate it to make it a bit more clear what the dependencies are that need to be keys for the remember. Or maybe even better, pull it out to a separate class that takes those dependencies as explicit constructor parameters. Right now the remember is missing some keys, and because the class captures the values outside the remember, I'm not even sure if providing the keys will be enough to make this work.

```suggestion
    val preloadModelProvider = remember(requestManager, requestBuilderModifier) {
      object : ListPreloader.PreloadModelProvider<DataTypeT> {
        var currentData: List<DataTypeT> = emptyList()
        override fun getPreloadItems(position: Int): MutableList<DataTypeT> {
          return mutableListOf(this.currentData[position])
        }

        override fun getPreloadRequestBuilder(item: DataTypeT): RequestBuilder<*> {
          return requestBuilderModifier(item, requestManager.asDrawable())
        }
      }
    }
```

Reply:
Ahh got it, ok done, thanks!